Disable menu on MINTEMP errors

DRracer opened the pull request:
Prevent the user from entering the main menu when there is a MINTEMP error. When the error fades out, the user must reset the printer to regain control.

DRracer committed:
Prevent the user from entering the main menu when there is a MINTEMP
error. When the error fades out, the user must reset the printer to
regain control.

DRracer committed:
Merge branch 'bugfixes' of github.com:DRracer/Prusa-Firmware into bugfixes

mkbel commented on Firmware/temperature.cpp:
```diff
@@ -2021,7 +2090,16 @@ void check_min_temp_heater0()
+		// we shall signalize, that MINTEMP has been fixed
+		// Code notice: normally the aam instance would have been placed here as static alert_automaton_mintemp aam, but
+		// due to stupid compiler that takes 16 more bytes.
+		aam[0].step(current_temperature[0], minttemp[0] + 50); // @@TODO tady pak dat TEMP_HYSTERESIS
```
@@TODO, it looks like something has been forgotten here.

mkbel commented on Firmware/temperature.cpp:
```diff
@@ -2013,6 +2035,53 @@ void check_max_temp()
+//! number of repeating the same state with consecutive step() calls
+//! used to slow down text switching
+struct alert_automaton_mintemp {
+	enum { ALERT_AUTOMATON_SPEED_DIV = 5 };
```
Doesn't need to be public, as it is never accessed from the outside.

mkbel commented on Firmware/temperature.cpp:
```diff
@@ -2013,6 +2035,53 @@ void check_max_temp()
+//! used to slow down text switching
+struct alert_automaton_mintemp {
+	enum { ALERT_AUTOMATON_SPEED_DIV = 5 };
+	uint8_t state, repeat = ALERT_AUTOMATON_SPEED_DIV;
```
Doesn't need to be public, as it is never accessed from the outside.

mkbel commented on Firmware/temperature.cpp:
```diff
@@ -2013,6 +2035,53 @@ void check_max_temp()
+	enum { ALERT_AUTOMATON_SPEED_DIV = 5 };
+	uint8_t state, repeat = ALERT_AUTOMATON_SPEED_DIV;
+	
+	void substep(uint8_t next_state){
```
Doesn't need to be public, as it is never accessed from the outside.

mkbel commented on Firmware/temperature.cpp:
```diff
@@ -2013,6 +2035,53 @@ void check_max_temp()
+		static const char m2[] PROGMEM = "MINTEMP fixed";
+		static const char m1[] PROGMEM = "Please restart";
+		switch(state){
+		case 0: // initial state - check hysteresis
```
Use meaningful names for states instead of comments. e.g. enum class

DRracer committed:
Fix for resume printing after filament runout. The previous
implementation first moved the nozzle back to last position over the
printout and then started heating, thus waiting a few seconds in the
resume position and possibly overheating the printout.

mkbel commented:
Contains unrelated work - commit Fix for resume printing after filament runout. The previous

mkbel commented on Firmware/Marlin_main.cpp:
```diff
@@ -1,9645 +1,9649 @@
-/* -*- c++ -*- */
```
It looks like you have changed 19294 lines in Marlin_main.cpp this is probably not intended, check line endings CR+LF vs LF

DRracer commented:
superseeded by another pull request, closing
